lib/src/context.dart

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2013-2016, the Clean project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+@JS()
+library react_dart_context;
+
+import 'dart:js_util';
+
+import 'package:js/js.dart';
+import 'package:react/react_client/react_interop.dart';
+import 'package:react/react_client.dart';
+
+/// The return type of [createContext], Wraps [ReactContext] for use in Dart.
+/// Allows access to [Provider] and [Consumer] Components.
+///
+/// __Should not be instantiated without using [createContext]__
+///
+/// __Example__:
+///
+///     ReactDartContext MyContext = createContext('test');
+///
+///     class MyContextTypeClass extends react.Component2 {
+///       @override
+///       final contextType = MyContext;
+///
+///       render() {
+///         return react.span({}, [
+///           '${this.context}', // Outputs: 'test'
+///         ]);
+///       }
+///     }
+///
+/// // OR
+///
+///     ReactDartContext MyContext = createContext();
+///
+///     class MyClass extends react.Component2 {
+///       render() {
+///         return MyContext.Provider({'value': 'new context value'}, [
+///           MyContext.Consumer({}, (value) {
+///             return react.span({}, [
+///               '$value', // Outputs: 'new context value'
+///             ]),
+///           });
+///         ]);
+///       }
+///     }
+///
+/// Learn more at: https://reactjs.org/docs/context.html
+class ReactDartContext {
```

Comment:
Idea: since this is now being namespaced, what about just calling it `Context`?

Reply:
I use `Context` for the over_react version, so i kept this one as `ReactReduxContext` just so there was no confusion between the two but i could, but should i?